plugins/woocommerce/includes/class-wc-emails.php

```diff
@@ -217,20 +225,36 @@ public function __construct() {
 	public function init() {
 		// Include email classes.
 		include_once dirname( __FILE__ ) . '/emails/class-wc-email.php';
+		$this->default_emails_paths = $this->default_emails_paths();
+       	$emails_data = apply_filters( 'woocommerce_before_email_classes', $this->default_emails_paths );
+    	foreach ( $emails_data as $class_name => $email_file_path ) {
+
+        		if ( ! empty( $email_file_path ) && ! empty( $class_name ) && file_exists( dirname( __FILE__ ) . $email_file_path ) ) {
+            		$this->emails[ $class_name ] = include dirname( __FILE__ ) . $email_file_path;
+       		}
+    	}
+		$this->emails = apply_filters( 'woocommerce_email_classes', $this->emails );	
+	}
 
-		$this->emails['WC_Email_New_Order']                 = include __DIR__ . '/emails/class-wc-email-new-order.php';
-		$this->emails['WC_Email_Cancelled_Order']           = include __DIR__ . '/emails/class-wc-email-cancelled-order.php';
-		$this->emails['WC_Email_Failed_Order']              = include __DIR__ . '/emails/class-wc-email-failed-order.php';
-		$this->emails['WC_Email_Customer_On_Hold_Order']    = include __DIR__ . '/emails/class-wc-email-customer-on-hold-order.php';
-		$this->emails['WC_Email_Customer_Processing_Order'] = include __DIR__ . '/emails/class-wc-email-customer-processing-order.php';
-		$this->emails['WC_Email_Customer_Completed_Order']  = include __DIR__ . '/emails/class-wc-email-customer-completed-order.php';
-		$this->emails['WC_Email_Customer_Refunded_Order']   = include __DIR__ . '/emails/class-wc-email-customer-refunded-order.php';
-		$this->emails['WC_Email_Customer_Invoice']          = include __DIR__ . '/emails/class-wc-email-customer-invoice.php';
-		$this->emails['WC_Email_Customer_Note']             = include __DIR__ . '/emails/class-wc-email-customer-note.php';
-		$this->emails['WC_Email_Customer_Reset_Password']   = include __DIR__ . '/emails/class-wc-email-customer-reset-password.php';
-		$this->emails['WC_Email_Customer_New_Account']      = include __DIR__ . '/emails/class-wc-email-customer-new-account.php';
-
-		$this->emails = apply_filters( 'woocommerce_email_classes', $this->emails );
+	/**
+	 * Get the default email classes.
+	 *
+	 * @return Array of default email classes and their paths.
+	 */
+	public function default_emails_paths() {
```

Comment:
For consistency with `get_emails()` I think it'd be best to use something like `get_default_emails()` or similar as name for this method.